## Online Lab of Pandas I

In this online lab, we will use the titanic data set.  The data is at "data\Titanic.tsv".  First, read in the data.

We are going to review the following knowledge points:

1. Slice dataframes
2. Use dataframes' built-in functions and attributes
3. Adding, droping and sorting columns

#### Ex1: Read in the dataset

In [1]:
import pandas as pd

titanic_df = pd.read_csv("Data/Titanic.csv")
titanic_df.head()

PassengerId  Pclass                                          Name     Sex  \
0          892       3                              Kelly, Mr. James    male   
1          893       3              Wilkes, Mrs. James (Ellen Needs)  female   
2          894       2                     Myles, Mr. Thomas Francis    male   
3          895       3                              Wirz, Mr. Albert    male   
4          896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female   

    Age  SibSp  Parch   Ticket     Fare Cabin Embarked  
0  34.5      0      0   330911   7.8292   NaN        Q  
1  47.0      1      0   363272   7.0000   NaN        S  
2  62.0      0      0   240276   9.6875   NaN        Q  
3  27.0      0      0   315154   8.6625   NaN        S  
4  22.0      1      1  3101298  12.2875   NaN        S

#### Ex2: Get the types of each column

In [2]:
titanic_df.dtypes

PassengerId      int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

#### Ex3: How many columns and rows are there in the data frame?

In [3]:
titanic_df.shape

(418, 11)

#### Ex4: Select out the PassengerId columns, and compute the maximum and minimum PassengerId

In [4]:
pid = titanic_df["PassengerId"]
max_pid = pid.max()
min_pid = pid.min()
max_pid, min_pid

(1309, 892)

#### Ex5: What is the average age of all customers?

In [5]:
avg_age = titanic_df["Age"].mean()
avg_age

30.272590361445783

#### Ex6: How many different passengear class are there

In [10]:
unique_pclass = len(titanic_df["Pclass"].unique())
unique_pclass

3

#### Ex7: Create a new column called "full_ticket"that contains both Embarked and Ticket number. For example, for passenger 892, this new column should be "Q-330911"

In [11]:
titanic_df["full_ticket"] = titanic_df["Embarked"] + "-" + titanic_df["Ticket"]
titanic_df.head()

PassengerId  Pclass                                          Name     Sex  \
0          892       3                              Kelly, Mr. James    male   
1          893       3              Wilkes, Mrs. James (Ellen Needs)  female   
2          894       2                     Myles, Mr. Thomas Francis    male   
3          895       3                              Wirz, Mr. Albert    male   
4          896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female   

    Age  SibSp  Parch   Ticket     Fare Cabin Embarked full_ticket  
0  34.5      0      0   330911   7.8292   NaN        Q    Q-330911  
1  47.0      1      0   363272   7.0000   NaN        S    S-363272  
2  62.0      0      0   240276   9.6875   NaN        Q    Q-240276  
3  27.0      0      0   315154   8.6625   NaN        S    S-315154  
4  22.0      1      1  3101298  12.2875   NaN        S   S-3101298

#### Ex8: Drop the new column that is created above.

In [12]:
titanic_df.drop("full_ticket", axis = 1, inplace=True)
titanic_df.head()

PassengerId  Pclass                                          Name     Sex  \
0          892       3                              Kelly, Mr. James    male   
1          893       3              Wilkes, Mrs. James (Ellen Needs)  female   
2          894       2                     Myles, Mr. Thomas Francis    male   
3          895       3                              Wirz, Mr. Albert    male   
4          896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female   

    Age  SibSp  Parch   Ticket     Fare Cabin Embarked  
0  34.5      0      0   330911   7.8292   NaN        Q  
1  47.0      1      0   363272   7.0000   NaN        S  
2  62.0      0      0   240276   9.6875   NaN        Q  
3  27.0      0      0   315154   8.6625   NaN        S  
4  22.0      1      1  3101298  12.2875   NaN        S

#### Ex9: Sort the original dataframe by age descendingly.

In [13]:
titanic_df.sort_values(by="Age", inplace=True, ascending=False)
titanic_df.head()

PassengerId  Pclass                                               Name  \
96           988       1  Cavendish, Mrs. Tyrell William (Julia Florence...   
81           973       1                                 Straus, Mr. Isidor   
236         1128       1                           Warren, Mr. Frank Manley   
305         1197       1  Crosby, Mrs. Edward Gifford (Catherine Elizabe...   
179         1071       1  Compton, Mrs. Alexander Taylor (Mary Eliza Ing...   

        Sex   Age  SibSp  Parch    Ticket      Fare    Cabin Embarked  
96   female  76.0      1      0     19877   78.8500      C46        S  
81     male  67.0      1      0  PC 17483  221.7792  C55 C57        S  
236    male  64.0      1      0    110813   75.2500      D37        C  
305  female  64.0      1      1    112901   26.5500      B26        S  
179  female  64.0      0      2  PC 17756   83.1583      E45        C